File: src/main/java/me/santio/uhc/states/ScoreboardState.java
package me.santio.uhc.states;

import lombok.Getter;
import me.santio.uhc.sidebars.LobbySidebar;
import me.santio.uhc.sidebars.Sidebar;

public enum ScoreboardState {
    LOBBY(new LobbySidebar()),
    IN_GAME(null),
    ADMIN(null);
    
    @Getter private final Sidebar sidebar;
    ScoreboardState(Sidebar sidebar) {
        this.sidebar = sidebar;
    }
}

File: src/main/kotlin/me/santio/uhc/states/ScoreboardState.kt
package me.santio.uhc.states

import me.santio.uhc.sidebars.LobbySidebar
import me.santio.uhc.sidebars.Sidebar

enum class ScoreboardState(val sidebar: Sidebar?) {
    LOBBY(LobbySidebar()), IN_GAME(null), ADMIN(null);
}